content/blog/2022-02-09-predictkube-scaler.md

```diff
@@ -0,0 +1,133 @@
+
+So, Dysnix started thinking about predictive autoscaling. To scale in time, you have to scale beforehand—or else, you'll have to deal with another problem that's familiar to each cloud inhabitant. The overprovision of resources is used to cover any traffic changes and results in bills and thrills connected with unexpected expenses. Can this process be more controllable?
+
+And when it comes to deciding how to wrap up this solution, Dysnix decided to rely on KEDA as it is the most universal and applicable component for application autoscaling in Kubernetes.
+
+KEDA is being used as a component on the client side of PredictKube that is responsible for transferring requests and scaling replicas.
+
+## Dysnix's PredictKube integrates with KEDA
+
+Dysnix has built [PredictKube](https://predictkube.com/), a solution that can be used as a KEDA scaler that is responsible for resource balancing, and an AI model that has learned to react proactively to patterns of traffic activity, to help with both in-time scaling and solving the problem of overprovision.
+
+![](/img/blog/predictkube-scaler/overprovision.png)
+
+The predictive autoscaling process is possible thanks to an AI model that observes the requests-per-second (RPS) or CPU values for a period of time during a project and then shows the trend for up to 6 hours. PredictKube used customer and open data sources (we used data sets like [HTTP NASA logs](ftp://ita.ee.lbl.gov/html/contrib/NASA-HTTP.html)) to train the model and be specific about the cloud data and traffic trends.
+
+With this tool, Dysnix wants to decrease costs on the projects, analyze the data about traffic more efficiently, use cloud resources more responsibly, and build infrastructures that are &quot;greener&quot; and more performative (with fewer downtimes and delays) than others.
+
+## How does PredictKube work?
+
+PredictKube works in two parts:
+
+1. **On the KEDA side**
+   The interface connects via API to the data sources about your traffic. PredictKube uses [Prometheus](https://prometheus.io/)—the industry standard for storing metrics. There, it anonymizes the data about the traffic on the client's side before sending it to the API, where the model then works with information that is completely impersonal.
+2. **On the AI model side**
+   Next, it is linked with a prediction mechanism—the AI model starts to get data about things that happen in your cloud project. Unlike standard rules-based algorithms such as Horizontal Pod Autoscaling (HPA), PredictKube uses Machine Learning models for time series data predicting, like CPU or RPS metrics.
+
+The more data you can provide to it from the start, the more precise the prediction will be. The 2+ weeks data will be enough for the beginning.
+
+![](/img/blog/predictkube-scaler/diagram.svg)
```

Comment:
Same here, let's resize and use a PNG/JPG

Reply:
converted to pnd